File: Removing Stars From a String.cpp
// https://leetcode.com/problems/removing-stars-from-a-string/description/


// use stack, TC:O(N), SC:O(N)
class Solution {
public:
    string removeStars(string s) {
        string res;
        for (char c : s) {
            // char c = s[i];
            if (c == '*')
                res.pop_back();
            else
                res.push_back(c);
        }
        return res;
    }
};

File: Removing Stars From a String.py
# https://leetcode.com/problems/removing-stars-from-a-string/description/


# stack, TC:O(N), SC:O(N)
def removeStars(s: str) -> str:
    stack = []
    for c in s:
        if c == '*':
            stack.pop()
        else:
            stack.append(c)
    return ''.join(stack)